build.cake

```diff
@@ -67,22 +68,59 @@ Task("Restore-NuGet-Packages")
     });
 });
 
-Task("Patch-Assembly-Info")
+Task("Run-GitVersion")
     .IsDependentOn("Restore-NuGet-Packages")
+    .IsDependentOn("Run-GitVersion-AppVeyor")
+    .IsDependentOn("Run-GitVersion-Local");
+
+
+Task("Run-GitVersion-AppVeyor")
+    .WithCriteria(AppVeyor.IsRunningOnAppVeyor)
     .Does(() =>
 {
-    var file = "./src/SolutionInfo.cs";
-    CreateAssemblyInfo(file, new AssemblyInfoSettings {
-        Product = "Cake",
-        Version = version,
-        FileVersion = version,
-        InformationalVersion = semVersion,
-        Copyright = "Copyright (c) Patrik Svensson, Mattias Karlsson, Gary Ewan Park and contributors"
+    GitVersion(new GitVersionSettings {
+        UpdateAssemblyInfoFilePath = "./src/SolutionInfo.cs",
+        UpdateAssemblyInfo = true,
+        OutputType = GitVersionOutput.BuildServer
     });
+    
+    version = EnvironmentVariable("GitVersion_MajorMinorPatch");
```

Comment:
On some CI Servers, because of the way GitVersion is ran, the generated Environment Variables are available immediately after calling GitVersion. In the case of AppVeyor, which we are using, this is not the case.  In theory, we could remove this, and only ever call GitVersion twice, but I thought it best to leave this in, as it provides more information about the "intent".